### PR TITLE
Default parser

### PR DESCRIPTION
**Pull request by epawlowska**

### CODE REVIEW

File: whylog/assistant/pattern_match.py [hunk 1]
```diff
@@ -2,4 +2,13 @@
 
 ParamGroup = namedtuple('ParamGroup', ['content', 'converter'])
 
-PatternMatch = namedtuple('PatternMatch', ['line_text', 'pattern', 'param_groups'])
```
Comment (ppolewicz) [inline whylog/assistant/pattern_match.py:5]: why?

File: whylog/assistant/regex_assistant/regex_match.py [hunk 2]
```diff
@@ -46,8 +47,10 @@ def update_by_regex(self, new_regex):
 
         if new_regex[-1] != '$':
             new_regex += '$'
+        if new_regex[0] != '^':
```
Comment (ppolewicz) [inline whylog/assistant/regex_assistant/regex_match.py]: this was just removed as a result of application of comment in another PR. Why add it back?

File: whylog/teacher/__init__.py [hunk 3]
```diff
@@ -18,8 +18,15 @@ def __init__(self, line_id, group_number_in_line):
 
 
 class TeacherParser(object):
```
Comment (ppolewicz) [inline whylog/teacher/__init__.py]: this is another namedtuple

File: whylog/teacher/__init__.py [hunk 4]
```diff
@@ -53,8 +62,27 @@ def add_line(self, line_id, line_object, effect=False):
...
-        self._parsers[line_id] = TeacherParser(line_object)
+
+        default_pattern_match = self.pattern_assistant.get_pattern_match(line_id)
+        default_pattern = default_pattern_match.pattern
+        default_groups = default_pattern_match.param_groups
+
+        default_name = self.config.propose_parser_name(
+            line_object.line_content, default_pattern, self.names_blacklist
+        )
+        defaulf_primary_key = [min(default_groups.keys())] if default_groups else []
```
Comment (ppolewicz) [inline whylog/teacher/__init__.py]: please rewrite that into a proper `if` statement

File: whylog/teacher/__init__.py [hunk 5]
```diff
@@ -154,16 +183,36 @@ def test_rule(self):
...
+        """
+        pattern_match = self.pattern_assistant.get_pattern_match(line_id)
+        teacher_parser = self._parsers[line_id]
+        pattern_type = self.pattern_assistant.TYPE
+        return UserParserIntent(
+            pattern_type, teacher_parser.name, pattern_match.pattern, teacher_parser.log_type,
+            teacher_parser.primary_keys, pattern_match.param_groups,
+            teacher_parser.line.line_content, teacher_parser.line.offset,
+            teacher_parser.line.line_source
+        )
```
Comment (ppolewicz) [inline whylog/teacher/__init__.py]: `# yapf: disable`

File: whylog/teacher/__init__.py [hunk 6]
```diff
@@ -53,8 +62,27 @@ def add_line(self, line_id, line_object, effect=False):
...
+        defaulf_primary_key = [min(default_groups.keys())] if default_groups else []
+        default_log_type_name = None
+
+        new_teacher_parser = TeacherParser(
+            line_object, default_name, defaulf_primary_key, default_log_type_name
+        )
+        self._parsers[line_id] = new_teacher_parser
+
+        self.names_blacklist.add(default_name)
```
Comment (ppolewicz) [inline whylog/teacher/__init__.py]: Why does it end up on a blacklist? Isn't it enough to store the name in one structure (self._parsers values and their name field)?

File: whylog/assistant/regex_assistant/regex_match.py (hunk 2 above)
Comment (epawlowska) [inline whylog/assistant/regex_assistant/regex_match.py]: Because I assumed that for each parser in config we want to have a regex with "^". Is it right?

Comment (ppolewicz) [inline whylog/assistant/regex_assistant/regex_match.py]: We use `re.match`, so still no.

File: whylog/tests/tests_teacher/__init__.py [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+            {
+                1: ParamGroup(
+                    content='2015-12-03 12:11:00',
+                    converter='to_date'
+                )
+            },
+            self.effect_front_input.line_content,
+            self.effect_front_input.offset,
+            self.effect_front_input.line_source,
+        )
```
Comment (ppolewicz) [inline whylog/tests/tests_teacher/__init__.py]: `# yapf: disable`

File: whylog/tests/tests_teacher/__init__.py [hunk 8]
```diff
@@ -68,5 +68,5 @@ def test_default_user_parser(self):
             self.effect_front_input.line_content,
             self.effect_front_input.offset,
             self.effect_front_input.line_source,
-        )
+            )  # yapf: disable
```
Comment (ppolewicz) [inline whylog/tests/tests_teacher/__init__.py]: fold is too far. Decrement by 4 spaces please.

**Pull request merged**